### ISSUE TITLE
False negative for NullableToStringCall rule in other class function calls.

### ISSUE THREAD

**Issue by poicooler** [detekt/detekt#6378]
## Expected Behavior
The NullableToStringCall rule should work the same when calling a function from another class.
```kotlin
class TestDetekt(val subClass: SubClass) {
    fun testDetekt(test: Int?) {
        test.toString() // detected
        test2(test.toString()) // detected
        subClass.test3(test.toString()) // not detected
    }

    private fun test2(value: String?) {}
}

class SubClass {
    fun test3(value: String?) {}
}
```

## Observed Behavior
The line `subClass.test3(test.toString())` is not flagged.

## Steps to Reproduce
You can use the code above to reproduce this issue.

## Context
We ended up having the string "null" instead of the null value.

## Your Environment
* Version of detekt used: 1.23.1
* Using detekt through detekt-maven-plugin version 1.23.1

**Comment by poicooler** [comment]
Might be related to #6349 but i do not think so.
